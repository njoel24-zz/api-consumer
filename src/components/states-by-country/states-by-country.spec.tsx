import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from "enzyme";

import StatesByCountry from './states-by-country.component';

describe('States by country', () => {
	const states = ['Alabama', 'Minnesota', 'Texas'];
	const getCitiesByState = jest.fn();

	it('renders component and compare with snapshot', () => {
		const component = renderer.create(<StatesByCountry states={states}/>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render states', () => {
		const wrapper = mount(
			<StatesByCountry states={states} />
		);
		const links = wrapper.find('a');
		expect(links.length).toBe(3);
	});

	it('should print states', () => {
		const wrapper = mount(
			<StatesByCountry states={states}/>
		);
		wrapper.find('a').forEach((singleLink) => {
			expect(singleLink.text()).not.toBeNull();
		});
	});

	it('should ask for city by clicking on a link', () => {
		const wrapper = mount(
			<StatesByCountry states={states} getCitiesByState={getCitiesByState}/>
		);
		const firstLink = wrapper.find("a:first-child");
		firstLink.simulate("click");
		expect(getCitiesByState).toBeCalled();
	});
});