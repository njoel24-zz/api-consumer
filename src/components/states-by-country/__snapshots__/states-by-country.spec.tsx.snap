// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`States by country renders component and compare with snapshot 1`] = `
<div>
  <a
    className=""
    onClick={[Function]}
  >
    Alabama
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    Minnesota
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    Texas
  </a>
</div>
`;
